intellij-idea.md

```diff
@@ -0,0 +1,182 @@
+* Add `type="war"`
+
+Also change `</webApplication>` to `</application>`.
+
+If the `server.xml` or any files in this directory change in the repository, make sure to update them accordingly.
+
+## Adding the CDS as a build and run configuration
+
+Now that we have set up all modules, we can finally add the build configurations. Setting up the CDS is the
+most cumbersome, but it is also the tool most other tools need, so it should be the first thing you set up.
+
+Click the `Run` menu item and select `Edit configurations...`. Click `+` and select `WebSphere Server`
+-> `Local`.  Name it `CDS` and uncheck `After launch` if you do not want your browser to launch after the
+CDS has started up. Use `https://localhost:8443/` as URL. Select `cds` under `WebSphere Server Settings` -> 
+`Server`. On the `Deployment` tab click the `+` button and select `Artifact...`. The CDS artifact will
+automatically be added. Make sure to check the `Use contest context root` checkbox and leave the context as `/`.
+Click the `Fix` button at the bottom if it complains about JMX administration.
+
+Now you can finally run and debug the CDS using the IntelliJ IDEA features. The CDS configuration can be found
+in the `wlp/usr/servers/cds` directory inside the Openliberty directory, so you can edit the `cdsConfig.xml` and
+other files there. You can use `File` -> `New` -> `Module from existing sources` and point to this directory to
+open it inside IntelliJ IDEA.
+
+## Adding other tools as a build an run configuration
+
+The other tools are way easier to run. Configuring each of them is similar with only small differences. First,
+the general process will be explained and below that the specific settings for each tool will be shown.
+
+In the `Run` menu, select `Edit configurations...`. Click the `+` button and select `Application`.
+Give the application a logical name. Under `Build and run` click the first empty dropdown and select the module
+you want to run. Click the Browse button inside the `Main class` input and select the correct class. If the tool
+requires program arguments, enter them in the `Program arguments` input field.
+
+When using macOS, click `Modify options` and select `Add VM options`. In the `VM options` field, type
+`-XstartOnFirstThread`.
+
+Click `OK` to add the tool.
+
+### Per tool settings
+
+* Balloon utility:
+  * Module: `BalloonUtility`
+  * Main class: `BalloonUtility`
+* Coach view:
+  * Module: `CoachView`
+  * Main class: `CoachView`
+  * Program arguments: `https://localhost:8443/api/contests/<id> public publ1c` where `<id>` is a configured contest ID
+* Floor generators:
+  * Module: `ContestUtil`
+  * Main class: `FloorGenerator<X>` with `<X>` the floor generator you want to use
+* Presentation admin:
+  * Module: `PresAdmin`
+  * Main class: `Admin`
+  * Program arguments: `https://localhost:8443/ admin adm1n`
+  * Note that the presentation admin requires a presentation client ZIP file inside a `present` directory in the same
```

Comment:
@deboer-tim is this true? If I run the admin without any ZIP present I see no presentations so I assume it is.
I always thought we only used this ZIP for updating the client to a newer version.

Reply:
Yes, it is required - the admin pulls the list of presentations from the client zip sitting on the CDS. The CDS build does this automatically so we normally don't notice.

Reply:
Ok. I couldn’t get IntelliJ to create a zip of another module during build so we just have to live with this for now